vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java

```diff
@@ -123,14 +143,25 @@ public ISpace getFullyLoadedSpace(String id) {
         return new ArrayList<>(spaceLinkDisplayRepo.findSpaceLinkDisplaysForSpace(spaceId));
     }
 
+    @Override
+    public List<IExternalLinkDisplay> getExternalLinkDisplays(String spaceId) {
+        return new ArrayList<>(externalLinkDisplayRepo.findExternalLinkDisplaysForSpace(spaceId));
+    }
+
     @Override
     public ISpaceLinkDisplay createSpaceLink(String title, ISpace source, float positionX, float positionY,
             int rotation, String linkedSpaceId, DisplayType displayType) throws SpaceDoesNotExistException {
         // we need this to fully load the space
+        ISpace target;
         source = spaceRepo.findById(source.getId()).get();
-        ISpace target = spaceRepo.findById(linkedSpaceId).get();
-        if (target == null) {
+        boolean isExists = spaceRepo.existsById(linkedSpaceId);
+        if(!isExists) {
             throw new SpaceDoesNotExistException();
+        } else {
+        target = spaceRepo.findById(linkedSpaceId).get();
```

Comment:
let's get an optional from 'findById' and call isPresent on it. The way you do it, this could still end up in an exception in case someone delete the space between the two calls you make to the repository (existsById and findById). There is also no reason for two DB calls.

Reply:
line 156 also needs to check if the space actually exist before getting it

Reply:
Maybe I can just use the getSpace(String id) method which does and checks all the conditions?

Reply:
We are doing a null check for space in AddSpaceLinkController and fetching the space before calling spaceManager. So isnt it assured that the space exists?

Reply:
sure that works.